app/models/piece.rb

```diff
@@ -11,13 +11,13 @@ def move_to!(new_x, new_y)
 		piece_to_capture = self.game.pieces.where(:x_pos => new_x, :y_pos => new_y).first
 		if piece_to_capture.present? && self.color.to_i != piece_to_capture.color.to_i
 			piece_to_capture.update_attributes(:x_pos => nil, :y_pos => nil) #captured pieces are nil thus not drawn and not clickable
-      self.update_attributes(:x_pos => new_x, :y_pos => new_y)
-      game.complete_turn
+			self.update_attributes(:x_pos => new_x, :y_pos => new_y)
```

Comment:
`game.complete_turn` needs to be called after `self.update_attributes`, otherwise it will stay on the same player's turn

Reply:
that explains alot